.travis.yml

```diff
@@ -1,11 +1,53 @@
+# Tell Travis which branches to build.
+branches:
+  except:
+    - /^(?i:dev)-.*$/
+
+# Tell Travis this is a Java project.
 language: java
-jdk:
-  - oraclejdk8
-  - openjdk7
+
+# Tell Travis this is a Java 7+8 project.
+matrix:
+  include:
+    - jdk: oraclejdk7
+    - jdk: oraclejdk8
+
+# Tell Travis we want container-based builds.
 sudo: false
+
+# Use the gradle wrapper for 2.7 instead of default 2.2
+before_install:
+  - chmod +x gradlew
+
+# Prepare the workspace.
+install:
+  - ./gradlew setupCIWorkspace -S
+
+# Build the Project.
+script:
+  - ./gradlew build -S
+
+# Push success to CurseForge for release.
+# after_success:
+#  - ./gradlew curse -S
+# Removed for ForgeGradle 2
+
+# Clean up after Gradle
+env:
+  global:
+    TERM=dumb
+
+# So the cache doesn't get rebuilt every time.
+before_cache:
+  - rm -f $HOME/.gradle/caches/modules-2/modules-2.lock
+  - rm -f $HOME/.gradle/caches/minecraft/ForgeVersion.json
+  - rm -f $HOME/.gradle/caches/minecraft/ForgeVersion.json.etag
+  - rm -rf $HOME/.gradle/caches/minecraft/deobfedDeps
+
+# Cache libs, as to improve speed.
 cache:
   directories:
-    - '$HOME/.m2'
```

Comment:
This prevents the cache from being constantly repackaged, in my experience, with the most possible amount of files preserved in the cache.